framework/src/utils/InputParameterWarehouse.C

```diff
@@ -17,7 +17,10 @@
 
 InputParameterWarehouse::InputParameterWarehouse() :
     Warehouse<InputParameters>(),
-    _name_to_shared_pointer(libMesh::n_threads())
+    _input_parameters(libMesh::n_threads()),
+    _system_to_index(libMesh::n_threads()),
+    _syntax_to_index(libMesh::n_threads()),
+    _tag_to_index(libMesh::n_threads())
```

Comment:
Is now the right time to dethread the InputParameters... or you want to do that separately?